### PR TITLE
nixos/doc/gnome: Document how python nautilus extensions work

### PR DESCRIPTION
**Pull request by doronbehar**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/desktop-managers/gnome.md [hunk 1]
```diff
@@ -116,6 +116,37 @@ Unfortunately, we lack a way for these to be managed in a completely declarative
...
+
+```nix
+environment.systemPackages = with pkgs; [
+  gnome.nautilus-python
+  nautilus-open-any-terminal
+];
+services.xserver.desktopManager.gnome.extraGSettingsOverridePackages = with pkgs; [
+  nautilus-open-any-terminal
+];
+```
```
Comment (jtojnar) [inline nixos/modules/services/x11/desktop-managers/gnome.md:141]: This should probably go to a separate section, as it is not specific to GNOME extensions. Though we will hopefully remove the option soon in favour of dconf module so perhaps it is not needed.

File: nixos/modules/services/x11/desktop-managers/gnome.md [hunk 2]
```diff
@@ -116,6 +116,37 @@ Unfortunately, we lack a way for these to be managed in a completely declarative
 So you have to enable them manually with an Extensions application.
 It is possible to use a [GSettings override](#sec-gnome-gsettings-overrides) for this on `org.gnome.shell.enabled-extensions`, but that will only influence the default value.
 
+## Nautilus Extensions {#sec-gnome-nautilus-extensions}
+
+Nautilus extensions can add additional functionality to Nautilus, usually via the right-click menu. Some of these extensions are dynamically loaded from `$NAUTILUS_4_EXTENSIONS_DIR` environment variable, which is automatically set if [](#opt-services.gnome.core-utilities.enable) is true (enabled by default if gnome is enabled).
```
Comment (jtojnar) [inline nixos/modules/services/x11/desktop-managers/gnome.md:121]: The environment variable is an implementation detail so I am not sure we should expose it. Ideally, we would drop it in favour of Nautilus wrapper package.

Comment (jtojnar) [inline nixos/modules/services/x11/desktop-managers/gnome.md:121]: GNOME should be capitalized.

Comment (jtojnar) [reply nixos/modules/services/x11/desktop-managers/gnome.md:121]: I wonder if we should have a Nautilus module in the meanwhile.

File: nixos/modules/services/x11/desktop-managers/gnome.md [hunk 3]
```diff
@@ -116,6 +116,37 @@ Unfortunately, we lack a way for these to be managed in a completely declarative
...
+
+Many Python based extensions can be found on the wild, and some of which are even packaged in Nixpkgs, (for example: `nautilus-open-any-terminal`). To load these extensions, you'll have to add the package that loads Python extensions for Nautilus:
+
+```nix
+environment.systemPackages = [
+  gnome.nautilus-python
```
Comment (jtojnar) [inline nixos/modules/services/x11/desktop-managers/gnome.md:127]: Missing `pkgs`.

File: nixos/modules/services/x11/desktop-managers/gnome.md [hunk 4]
```diff
@@ -116,6 +116,37 @@ Unfortunately, we lack a way for these to be managed in a completely declarative
...
+environment.systemPackages = with pkgs; [
+  gnome.nautilus-python
+  nautilus-open-any-terminal
+];
+services.xserver.desktopManager.gnome.extraGSettingsOverridePackages = with pkgs; [
+  nautilus-open-any-terminal
+];
+```
+
+In fact, any package that includes a python script at `/share/nautilus-python/extensions`, will provide a nautilus Python extension. Other examples include:
```
Comment (jtojnar) [inline nixos/modules/services/x11/desktop-managers/gnome.md:143]: Maybe we should also have a `nautilus-python` module that adds this to `pathsToLink`.

**Review by doronbehar — COMMENTED** [review]
Thanks for the review @jtojnar. Regarding many of your comments that prospect about improving the module - I really think they are good ideas, but in the meantime the current behavior is not documented so maybe we can postpone them to a later time? Perhaps after this PR, users will read the current state of things and will consider improving the situation via a `nautilus` wrapper and a module, and `dconf` module.

**Comment by doronbehar** [comment]
Just fixed the phrasing of the commit title...

File: nixos/modules/services/x11/desktop-managers/gnome.md [hunk 5]
```diff
@@ -155,6 +178,21 @@ services.xserver.desktopManager.gnome = {
 };
 ```
 
+Adding the `nautilus-open-terminal-any` extension including `gsettings` schemas:
+
+```nix
+# Adds to `environment.pathsToLink` the path: `/share/nautilus-python/extensions`
+# needed for nautilus Python extensions to work.
+servuces,gnome.core-utilities.enable = true;
```
Comment (dvonessen) [inline nixos/modules/services/x11/desktop-managers/gnome.md:186]:
```suggestion
services.gnome.core-utilities.enable = true;
```

Comment (doronbehar) [reply nixos/modules/services/x11/desktop-managers/gnome.md:186]: Thanks! Done.

**Comment by doronbehar** [comment]
Closed PR by mistake, and I still want to reach an agreement upon it and merge it. Now rebased it upon current `master`.

**Comment by jtojnar** [comment]
Okay, here is my alternative #319535

**Comment by doronbehar** [comment]
Closed by mistake.

**Comment by doronbehar** [comment]
Trying in a new PR from a new branch at https://github.com/NixOS/nixpkgs/pull/352307 (Due to weird CI errors).

**Pull request closed without merging**